bz-manage.sh

```diff
@@ -381,4 +381,220 @@ get_bz_id_summary() {
+  gh_pr_info=$(gh pr view "$gh_pr" --json title,body --jq .title,.body)
+  gh_pr_desc=$(echo "$gh_pr_info" | sed 1d | sed -e 's/[[:space:]]*$//' -e '/^$/d')
+  gh_pr_title=$(echo "$gh_pr_info" | sed -n 1p)
+  if [[ "$gh_pr_title" =~ ^feat.* ]]; then
+    gh_pr_type="Enhancement"
+  elif [[ "$gh_pr_title" =~ ^fix.* ]]; then
+    gh_pr_type="Bug Fix"
+  else
+    gh_pr_type="Enhancement"
+  fi
+}
+
+bz_add_doc_text() {
+  gh_pr=$1
+  bz=$2
+  get_pr_info "$gh_pr"
+  comment="Updating Doc Type and Text with info from the attached PR $gh_pr"
+  echo "BZ#$bz: $comment"
+  bugzilla modify --private \
+    --field cf_doc_type="$gh_pr_type" \
+    --field cf_release_notes="$gh_pr_desc" \
+    --comment "$comment" \
+    "$bz"
+}
+
+print_doc_text() {
+  doc_type=$1
+  doc_text=$2
+  echo "~~~"
+  echo "Doc Type:
+$doc_type"
+  echo -e "Doc Text:
+$doc_text"
+  echo "~~~"
+}
+
+bz_mod_doc_text() {
+  gh_pr=$1
+  bz=$2
+  query_file=$3
+  get_pr_info "$gh_pr"
+  doc_text="$(jq -r "select(.id == $bz) | .cf_release_notes" "$query_file" | sed -e 's/[[:space:]]*$//' -e '/^$/d')"
+  if [ -z "$doc_text" ]; then
+    comment="Updating Doc Type and Text with info from the attached PR $gh_pr"
+    echo "BZ#$bz: $comment"
+    bugzilla modify --private \
+      --field cf_doc_type="$gh_pr_type" \
+      --field cf_release_notes="$gh_pr_desc" \
+      --comment "$comment" \
+      "$bz"
+  else
+    if [ "$doc_text" != "$gh_pr_desc" ]; then
+      echo "BZ#$bz has Doc Text that differs from GitHub PR description"
+      echo "The current Doc Type and Text:"
+      print_doc_text "$doc_type" "$doc_text"
+      echo ""
+      echo "Doc Type and Text suggested from the GH PR:"
+      print_doc_text "$gh_pr_type" "$gh_pr_desc"
+      while read -r -p 'Do you want to update the BZ with the suggested Doc Text and Type (y/n)? (default: n) ' update_bz </dev/tty; do
+        if [ "${update_bz:-n}" = n ]; then
+          return
+        elif [ "${update_bz:-n}" = y ]; then
+          comment="Updating Doc Type and Text with info from the attached PR $gh_pr"
+          echo "BZ#$bz: $comment"
+          bugzilla modify --private \
+            --field cf_doc_type="$gh_pr_type" \
+            --field cf_release_notes="$gh_pr_desc" \
+            --comment "$comment" \
+            "$bz"
+          return
+        else
+          echo "$update_bz is not y or n"
+        fi
+      done
+    else
+      echo "BZ#$bz already has Doc Text that equals GitHub PR description"
+    fi
+  fi
+}
+
+process_count_prs() {
+  bz="$1"; shift
+  func="$1"; shift
+  echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+  jq_count_prs="select(.id == $bz)\
+| [.external_bugs[]\
+| select((.type.type == \"GitHub\") and (.ext_bz_bug_id | contains(\"pull\")))]\
+| length"
+  gh_pr_count=$(jq -r "$jq_count_prs" "$query_file")
+  if [ "$gh_pr_count" -eq 1 ]; then
+    # eval to re-evaluate arguments for the $func function
+    # shellcheck disable=SC2294
+    eval "$func" "$@"
```

Comment:
not sure why you need to use `eval` here - what does `eval` do vs. just doing `"$func" "$@"`?

Reply:
Running with debug with eval:
```
+ eval bz_mod_doc_text '$gh_pr $bz $query_file'
++ bz_mod_doc_text https://github.com/linux-system-roles/auto-maintenance/pull/281 2219317 .query_file.json
```
Without:
```
+ bz_mod_doc_text '$gh_pr $bz $query_file'
+ gh_pr='$gh_pr $bz $query_file'
bz-manage.sh: line 425: $2: unbound variable
```

Reply:
I tried to explain it in a comment above somehow, let me elaborate more here.
To `process_count_prs` with the last argument, I provide a list of arguments that I want to run with the $func function.
So I am running `process_count_prs "$bz" bz_mod_doc_text '$gh_pr $bz $query_file'` and then run `eval "$func" "$@"`.
If I use it like `process_count_prs "$bz" bz_mod_doc_text "$gh_pr $bz $query_file"` - pass last arg in double quotes, then `$gh_pr $bz $query_file` are populated with the initial values. But `process_count_prs` must re-evaluate `$gh_pr` value based on what PR serial number user provides.

This is needed because bz_add_doc_text takes 2 arguments - `$gh_pr $bz`, and `bz_mod_doc_text` takes 3 arguments - `$gh_pr $bz $query_file`.

As a workaround, in `process_count_prs`, I can also run func simply as `"$func" "$gh_pr" "$bz" "$query_file"`
It would naturally work for `bz_mod_doc_text` and won't break anything for `bz_add_doc_text` too, but would be hard to troubleshoot is something gets changed.

Reply:
I would prefer to find a solution that doesn't use `eval` but it may be unavoidable in this case.

What confuses me is that in https://github.com/linux-system-roles/auto-maintenance/pull/281/files#diff-e05dfb61babe0363643fe39c9ccdebea9f2e2fc34ad9c95b6fdf69b3eb748f60R529 you grab `gh_pr`, but you don't necessarily use it - in the case where `"$gh_pr_count" -gt 1` you may have multiple different values of `gh_pr`, and you want to use the value which is set in the loop at 496.

So I think this is ok